File: Leetcode/9. Palindrome Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0){
            return false;
        }
        int num = x;
        long reversed = 0;
        while (num > 0){
            int lastDigit = num % 10;
            reversed = reversed * 10 + lastDigit;
            num /= 10;
        }
        if (x == reversed){
           return true;
        }
        else {
            return false;
        }
    }

};

File: Leetcode/9. Palindrome Number.py
class Solution(object):
    def isPalindrome(self, x):
        if x < 0:
            return False
        num = x
        reverse = 0
        while num > 0:
            lastdigit = num % 10
            reverse = reverse * 10 + lastdigit
            num = num / 10
        if reverse == x:
            return True
        else:
            return False
    # return str(x) == str(x)[::-1] # This would be the one liner solution
        